### PR TITLE
fix: guard against NaN/Infinite values to prevent crashes during crop and rotation

### PR DESCRIPTION
**Pull request by guoyingtao**
Add defensive checks throughout the crop pipeline to prevent two categories of crashes reported on iOS/iPadOS 26.0+:

1. "Double value cannot be converted to Int because it is either infinite or NaN" in CGImageExtensions.swift when outputSize contains NaN/Inf from division by zero imageViewSize or zoomScale.

2. CALayerInvalidGeometry when setting cropAuxiliaryIndicatorView.frame with NaN values during 90° rotation animation.

Key changes:
- Validate outputSize, cropSize, imageViewSize before CGContext creation
- Guard getOutputCropImageSize and cropWithPerspective against zero divisors
- Reject NaN cropBoxFrame in CropViewModel setter to stop propagation
- Guard handleCropBoxFrameChange against invalid frame geometry
- Protect adjustWorkbenchView and updatePositionFor90Rotation from NaN sizes
- Guard getBoundZoomScale against zero-size bounds/container
- Replace fatalError with assertionFailure in CoreGraphicsExtensions so NaN values degrade gracefully in production instead of crashing

## Summary by CodeRabbit

* **Bug Fixes**
  * Added defensive validation across cropping, layout, and image-transformation flows to reject non-finite or non-positive geometry.
  * Prevented applying invalid state (zoom, offsets, masks, layout) and replaced hard crashes with safe fallbacks/assertions for improved stability during edge-case geometry and rotations.
  * Image transformation and perspective-crop now fail fast when inputs or computed sizes are invalid.
* **Tests**
  * Added a unit test ensuring invalid crop-frame values are rejected and do not trigger change callbacks.

### CODE REVIEW

**Comment by guoyingtao** [comment]
@coderabbitai review

File: Sources/Mantis/Extensions/CGImageExtensions.swift [hunk 1]
```diff
@@ -23,6 +23,13 @@ extension CGImage {
                           outputSize: CGSize,
                           cropSize: CGSize,
                           imageViewSize: CGSize) throws -> CGImage? {
+        guard outputSize.width > 0, outputSize.height > 0,
+              outputSize.width.isFinite, outputSize.height.isFinite,
+              cropSize.width > 0, cropSize.height > 0,
+              imageViewSize.width > 0, imageViewSize.height > 0 else {
+            return nil
```
[the remark this answers is not in the document]
Comment (guoyingtao) [reply Sources/Mantis/Extensions/CGImageExtensions.swift:32]: Fixed

File: Sources/Mantis/CropView/CropViewModel.swift [hunk 2]
```diff
@@ -61,6 +61,11 @@ final class CropViewModel: CropViewModelProtocol {
     
     var cropBoxFrame = CGRect.zero {
         didSet {
+            guard cropBoxFrame.origin.x.isFinite, cropBoxFrame.origin.y.isFinite,
+                  cropBoxFrame.width.isFinite, cropBoxFrame.height.isFinite else {
+                cropBoxFrame = oldValue
+                return
```
[the remark this answers is not in the document]
Comment (guoyingtao) [reply Sources/Mantis/CropView/CropViewModel.swift]: Fixed
